rc/base/lua.kak

```diff
@@ -11,11 +11,12 @@ hook global BufCreate .*[.](lua) %{
 # Highlighters
 # ‾‾‾‾‾‾‾‾‾‾‾‾
 
-add-highlighter shared/ regions -default code lua \
-    string  '"'      (?<!\\)(\\\\)*"  '' \
-    string  "'"      (?<!\\)(\\\\)*'  '' \
-    comment '--\[\[' '\]\]'           '' \
-    comment '--'     '$'              '' \
+add-highlighter shared/ regions -default code -match-capture lua \
+    string  '"'           (?<!\\)(\\\\)*" '' \
+    string  "'"          (?<!\\)(\\\\)*'  '' \
+    string  '\[(=*)\['   '\](=*)\]'       '' \
+    comment '--\[(=*)\[' '\](=*)\]'       '' \
```

Comment:
Won't this highlight weird cases like this incorrectly?

``` lua
--[=[
	Hello, this is a comment!
]]
]=]--

function test()
	print("Hello")
end
```
I think the number of `=` somehow needs to be guarded to be the same on the opening and on the closing side, but I'm not sure how it would be implemented.

Reply:
![image](https://user-images.githubusercontent.com/356714/36116892-c07faa5c-0fec-11e8-9411-105e01f0e2c3.png)

This is handled by the `-match-capture`. The `(=*)` is capturing the number of `=` entries.

Reply:
Oh, right, I've missed that part. That's pretty neat then :)